web/pdf_viewer.js

```diff
@@ -1226,7 +1232,13 @@ class PDFViewer {
     if (this.pagesCount === 0) {
       return;
     }
-    this.update();
+
+    const visible = this._getVisiblePages();
+    for (const { view, visibleArea } of visible.views) {
+      view.updateVisibleArea(visibleArea);
+    }
+
+    this.update(visible);
```

Comment:
Maybe I'm not understanding the code, but why place the `updateVisibleArea`-call here rather than just placing all of this new code in the `update`-method?

Note that `update` is invoked in other spots as well, e.g. when changing scroll/spread modes (which may change the visible page area), and I'd assume that we always want to call `updateVisibleArea`!?

Reply:
I didn't realize that the other callers of `.update()` can actually cause a change in the visible area. I was focusing on `.refresh()`, where _probably_ the area is going to remain the same.